### PR TITLE
link cta widget

### PR DESCRIPTION
**Pull request by forresto**
- [x] set up widget
- [x] import `iframe src` or `a href` from html string
- [x] add... menu
- [x] verify block survives html-flatten

### CODE REVIEW

File: test/schema/block-meta.js [hunk 1]
```diff
@@ -61,6 +61,26 @@ describe('BlockMeta', function () {
...
+
+    it('gives expected html out', function () {
+      const html = BlockMeta.cta.makeHtml(cta.metadata)
+      expect(html).to.equal(
+        '<a href="http://fff" data-role="cta" >label y&ouml;</a>'
+      )
+    })
+    // it('gives html that survives html-flatten', function (done) {
+    //   survivesHtmlFlatten(cta, done)
+    // })
```
Comment (forresto) [inline test/schema/block-meta.js]: TODO!

**Comment by forresto** [comment]
ed-cta

**Comment by forresto** [comment]
Do average people know what "Call To Action" is?

add cta

File: src/components/widget-cta.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import ButtonOutline from 'rebass/dist/ButtonOutline'
+
+import TextareaAutosize from './textarea-autosize'
+import {isUrlOrBlank} from '../util/url'
+import {widgetLeftStyle} from './rebass-theme'
+
+// Gets src or href from iframe or a
+// http://www.regexpal.com/ test string:
+// asfd  dfadf ads <iframe  a="ffff" src="fooo" sss></iframe> <a hRef='http://,,,'>thing!</a> asdf
+const extractLink = /<(iframe|a)\s+[^>]*(?:src|href)=["'](\S+)["'][^>]*>/i
```
Comment (forresto) [inline src/components/widget-cta.js]: Somebody please sanity check this regex? 😨

File: test/components/widget-cta.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      const extract = extractLink('<iframe src="https://embed..."></iframe>')
+      expect(extract).to.deep.equal({tag: 'iframe', link: 'https://embed...'})
+    })
+    it('gives expected tag and link with iframe html string with other text', function () {
+      const extract = extractLink('abc <iframe src="https://embed..."></iframe> abc')
+      expect(extract).to.deep.equal({tag: 'iframe', link: 'https://embed...'})
+    })
+    it('gives expected tag and link, case-insensitive', function () {
+      const extract = extractLink('<IFRAME SRc="https://embed..."></IFRAME>')
+      expect(extract).to.deep.equal({tag: 'iframe', link: 'https://embed...'})
```
Comment (forresto) [inline test/components/widget-cta.js:18]: @jonnor writing tests helped me fix this case 👍

File: src/components/widget-cta.js (hunk 2 above)
Comment (pvdz) [reply src/components/widget-cta.js]: In theory (so this is mostly edge cases and super edge cases);
- iirc there can be spaces between `<` and the tag name
- you may be matching `src` that's part of a different payload or even custom attribute (`<iframe foosrc="..">` or `<iframe onclick="foo.src='bar';">`) 
- the value may not be quoted (`<iframe src=foo.html>`)
- there can be whitespace next to the equal sign (I think even before, but certainly after) that may be ignored (stuff like `<iframe src= "foo">` and `<iframe src ="foo">` (?) and even `<iframe src= foo>` iirc). The spec is weird like that.
- the `(\S+)` should be some kind of backreference to the quote parsed earlier, but at least have a `?` to be non-greedy otherwise it will skip quotes as long as another quote can be found (`<iframe src="foo" and="bar">` would lead to `foo" and ="bar`), so use `(\S+?)` to make the `+` be "non-greedy".

I hope `extractLink` is sanitized afterwards because there can be a lot of crap inside of it. But that's beyond the scope of the question or the PR.
